pkg/controller/operators/catalog/operator.go

```diff
@@ -1033,6 +1044,96 @@ func (o *Operator) createInstallPlan(namespace string, subs []*v1alpha1.Subscrip
 	return reference.GetReference(res)
 }
 
+func (o *Operator) checkBundleLookups(plan *v1alpha1.InstallPlan) (bool, error) {
+	for _, bundleLookup := range plan.Status.BundleLookups {
```

Comment:
Over the course of our work on OLM, we've developed an idiomatic way of reconciling resources:

1. Build state from cluster
2. Inform cluster action from state; or
3. Update resource status from state

I think that we can run into trouble if we jump straight to 2. without first performing 1., or by coupling 2. and 3.

In this case, do you think we can:

- build the state before actioning on it; i.e. Check for `ConfigMaps`/`Jobs` before creating them
- update the `InstallPlan` object once per reconciliation

Reply:
This is a good plan - we just need a deterministic way to name the configmaps/jobs from the installplan. InstallPlanName + image name would make sense, or a hash of that info.